src/randomizer.py

```diff
@@ -1941,36 +1941,44 @@ def visit_Assignment(self, node: ca.Assignment) -> None:
 def perm_chain_assignment(
     fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
 ) -> None:
-    """Combine two consecutive assignments into one chain assignment."""
-    cands: List[Tuple[int, Block]] = []
+    """Combine two assignments into one chain assignment."""
+    cands: List[Tuple[int, int, Block]] = []
 
     def rec(block: Block) -> None:
         statements = ast_util.get_block_stmts(block, False)
-        for i, (stmt, next_stmt) in enumerate(zip(statements, statements[1:])):
+
+        for i, stmt in enumerate(statements):
             if (
-                region.contains_node(stmt)
-                and region.contains_node(next_stmt)
-                and isinstance(stmt, ca.Assignment)
-                and isinstance(next_stmt, ca.Assignment)
-                and (
-                    ast_util.equal_ast(stmt.lvalue, next_stmt.rvalue)
-                    or ast_util.equal_ast(stmt.rvalue, next_stmt.rvalue)
-                )
-                and not ast_util.is_effectful(next_stmt.rvalue)
+                not isinstance(stmt, ca.Assignment)
+                or not region.contains_node(stmt)
+                or ast_util.is_effectful(stmt.rvalue)
             ):
-                cands.append((i, block))
+                continue
+            for j, next_stmt in enumerate(statements[i + 1 : i + 5], i + 1):
+                if (
+                    region.contains_node(stmt)
+                    and region.contains_node(next_stmt)
+                    and isinstance(stmt, ca.Assignment)
+                    and isinstance(next_stmt, ca.Assignment)
+                    and (
+                        ast_util.equal_ast(stmt.lvalue, next_stmt.rvalue)
+                        or ast_util.equal_ast(stmt.rvalue, next_stmt.rvalue)
+                    )
+                    and not ast_util.is_effectful(next_stmt.rvalue)
+                ):
+                    cands.append((i, shift, block))
```

Comment:
`shift` isn't defined. Let's do `j` and adjust the logic below

Reply:
in this case won't j still represent a shift value?

an offset from i, rather than an absolute index,  therefore the logic below should stay the same?

I definitely could be confused here though